tutorials/haskell/golden-testing-in-haskell-with-hspec-gold/tutorial.md

```diff
@@ -0,0 +1,190 @@
+---
+title: Golden testing in Haskell with hspec-golden
+published: 2019-08-10
+tags: haskell, golden testing, snapshot testing, expect testing
+libraries: hspec-golen-0.1.0.0
+language: haskell
+author: Adrian Vera
+author-name: Adrian Vera
+twitter-profile: adrian_med
+github-profile: medAdrian
+description: In this tutorial we will implement golden tests in Haskell using hspec-golden a super set of the known test framework hspec. A golden test is an Input/Output action that records the result of a test into a file known as the "Golden File".
+---
+
+Golden testing (also known as Golden master testing or Snapshot testing) is pretty similar to unit testing, except that the test output is stored in a separate file known as the "Golden File".
+
+This kind of testing will help us verify modifications made to a reference version of our SUT(Subject Under Test) and validate that it has not modified its
+behaviour in an incorrect way. Thus giving us some safety when extending and refactoring code that doesn't have proper tests.
+
+When we write golden tests we build up knowledge of what the code is doing, this is very useful when we want to re-write it,
+we can run our test to find out immediately whether we've modified the desired behaviour (we start to see our tests in a different way
+since the purpose of golden testing is to document your system's behaviour, not to check for the wished one).
+
+Once we adopt this perspective, we have a completely different frame for understanding automated tests in general,
+we can see tests as descriptions of what we have rather than statements of correctness.
+We can review our tests periodically to tighten up their conditions as we decide what the behavior should be at each level of our systems.
+
+##The Main idea
```

Comment:
@MedAdrian Do you think it is necessary a separated section to quote `A Golden File ...`? I think this can be above the introduction. What do you think?

Reply:
I think there's no need just for this quote, i added it as part of the intro.